platforms/jvm/language-groovy/src/integTest/groovy/org/gradle/groovy/compile/GroovyCompileProblemsIntegrationTest.groovy

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.groovy.compile
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class GroovyCompileProblemsIntegrationTest extends AbstractIntegrationSpec {
+
+    def setup() {
+        enableProblemsApiCheck()
+
+        buildFile << """\
+            apply plugin: 'groovy'
+
+            dependencies {
+                implementation localGroovy()
```

Comment:
❌ Let's keep the test clean. `localGroovy()` is for developing Gradle plugins and we test plain Groovy compilation here.

```suggestion
                implementation 'org.codehaus.groovy:groovy-all:x.y.z'
```